_posts/2017/2017-04-19-build-a-multiple-repository-pull-request-test.md

```diff
@@ -0,0 +1,161 @@
+---
+layout: post
+title: "Build a multiple repository pull request test"
+date: "2017-04-19 10:00:00"
+image: blog-header/challenge-cube.jpg
+categories: tech-stories
+authors: ["Florian Z."]
+---
+
+Did you ever face the problem of building a feature over multiple repositories?
+If you start testing the single repository, you may have unknown side effects when your code is rolled out.
+On the other hand, if you start testing the repositories together, you may have an inconsistent state of one of your code changes.
+This will lead to either errors in the test suite or wrong testing results.
+Broken tests, wrong test results and unknown side effects cause a pipeline to misses its validity.
+Developing effort and the risk of loss of quality increases significantly, but you can solve the situation with building up a multiple repositories pull request test.
+
+### 1. How do we prepare the test?
+The solution is as simple as complicated: Your Pull request has to know, which code change belongs to a specific feature.
+While using three different repositories, how can you achieve to test all changes at once?
+Three things are necessary:
+* GitHub Organizational Plugin,
+* Pipeline plugin,
+* naming convention.
+
+### 2. Why do we need those things?
+You can use the [Github Organizational Folder Plugin](https://wiki.jenkins-ci.org/display/JENKINS/GitHub+Organization+Folder+Plugin) to check your repositories for new Pull request.
+This plugin will check every affected repository for changes and activate the Jenkinsfile in the main directory, if a change was made.
+
+You can create a [Jenkinsfile](https://jenkins.io/doc/book/pipeline/jenkinsfile/) at a repository, which will be triggered after a Pull request.
+The task of this Jenkinsfile is mainly to check, if branches with the same name from the same developer in other repositories exists and will use them for the PR tests, if a change was detected.
```

Comment:
... in other repositories exist and then uses them for the PR tests, if a change has been detected.